src/main.zig

```diff
@@ -2419,20 +2419,35 @@ pub fn cmdBuild(gpa: *Allocator, arena: *Allocator, args: []const []const u8) !v
         try updateModule(gpa, comp, null, .none);
         try comp.makeBinFileExecutable();
 
+        // To be able to have multiple instances of `zig build` running on Windows with a difference in
+        // the source of the `build.zig` file, we have to copy the executable to a name that includes
+        // a hash of the content of the `build.zig` file and then execute that file, instead of the
+        // original build executable.
+        const build_zig_file = try build_directory.handle.openFile(build_zig_basename, .{});
+        defer build_zig_file.close();
+
+        var bin_digest: [Cache.bin_digest_len]u8 = undefined;
+        try Cache.hashFile(build_zig_file, &bin_digest);
+
+        var file_name: [Cache.hex_digest_len]u8 = undefined;
+        _ = std.fmt.bufPrint(&file_name, "{x}", .{bin_digest}) catch unreachable;
+
+        const temp_file_basename = try std.zig.binNameAlloc(arena, .{
+            .root_name = &file_name,
+            .target = target_info.target,
+            .output_mode = .Exe,
+        });
+
+        const dir = comp.bin_file.options.emit.?.directory.handle;
+        _ = try fs.Dir.updateFile(dir, exe_basename, dir, temp_file_basename, .{});
```

Comment:
Since you're not updating anything:

```suggestion
        try dir.copyFile(exe_basename, dir, temp_file_basename, .{});
```